src/components/Header/renderHeaderItems/index.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+
+import { Dropdown } from '../../Dropdown';
+import { HeaderItem } from '../component';
+import { Item } from '../styled';
+
+import { StyledDropdown, StyledDefaultTarget } from './styled';
+
+export const renderHeaderItems = (items: HeaderItem[], isDropdownItem = false) => {
+  return items.map((it, index) => {
+    const { children, target, isStyled, ...otherItemProps } = it;
+
+    if (!children) {
```

Comment:
I think you can invert this `if` to avoid nesting.

```tsx
if (children) {
  return <StyledDropdown ………>
}

if (isDropdownItem) {
  return …
}

if (isStyled) {
  return …
}

return <Item …>
```